manager/controllers/app/plotter_generator.go

```diff
@@ -130,148 +131,106 @@ func (p *PlotterGenerator) AddFlowInfoForAsset(item DataInfo, appContext *app.Fy
 		Format:     item.DataDetails.Interface.DataFormat,
 	}
 
-	assets[datasetID] = app.AssetDetails{
+	assets[item.Context.DataSetID] = app.AssetDetails{
 		AdvertisedAssetID: "",
 		DataStore:         *sourceDataStore,
 	}
 
 	// DataStore for destination will be determined if an implicit copy is required
 	var sinkDataStore *app.DataStore
 
-	selectors, err := p.SelectModules(&item, appContext)
-	if err != nil {
-		return err
+	solutions := p.FindPaths(&item, appContext)
+	if len(solutions) == 0 {
+		return errors.New("Data path could not be constructed")
 	}
 	p.Log.V(0).Info("Generating a plotter")
-	readSelector := selectors[app.Read]
-	copySelector := selectors[app.Copy]
-	// sanity
-	if readSelector == nil && copySelector == nil {
-		return errors.New("Can't generate a plotter - no modules available")
-	}
-	var copyCluster, readCluster string
-	if copySelector != nil {
-		p.Log.Info("Found copy module " + copySelector.GetModule().Name + " for " + datasetID)
-		// copy should be applied - allocate storage
-		if sinkDataStore, err = p.GetCopyDestination(item, copySelector.Destination, copySelector.StorageAccountRegion); err != nil {
-			p.Log.Info("Allocation failed: " + err.Error())
-			return err
-		}
-		var copyDataAssetID = datasetID + "-copy"
-		actions := createActionStructure(copySelector.Actions)
-		if len(item.Configuration.ConfigDecisions[app.Copy].Clusters) > 0 {
-			copyCluster = item.Configuration.ConfigDecisions[app.Copy].Clusters[0]
-		} else {
-			msg := "Coud not determine the cluster for copy"
-			p.Log.Info(msg)
-			return errors.New(msg)
-		}
-		// The default capability scope is of type Asset
-		scope := copySelector.ModuleCapability.Scope
+	selection := solutions[0]
```

Comment:
There could be more than one possible path. Yet, it is not exactly a random choice since the paths are sorted by their length. Thus, we'll choose the shortest path when possible, for example: read over copy+read.